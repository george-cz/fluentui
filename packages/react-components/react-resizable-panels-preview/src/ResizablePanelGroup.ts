export * from './components/ResizablePanelGroup/index';
