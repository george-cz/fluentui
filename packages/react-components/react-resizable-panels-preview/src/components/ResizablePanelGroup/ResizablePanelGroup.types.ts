import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';

export type ResizablePanelGroupSlots = {
  root: Slot<'div'>;
};

/**
 * ResizablePanelGroup Props
 */
export type ResizablePanelGroupProps = ComponentProps<ResizablePanelGroupSlots> & {};

/**
 * State used in rendering ResizablePanelGroup
 */
export type ResizablePanelGroupState = ComponentState<ResizablePanelGroupSlots>;
// TODO: Remove semicolon from previous line, uncomment next line, and provide union of props to pick from ResizablePanelGroupProps.
// & Required<Pick<ResizablePanelGroupProps, 'propName'>>
