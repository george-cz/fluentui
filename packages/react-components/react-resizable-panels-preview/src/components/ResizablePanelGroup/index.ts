export * from './ResizablePanelGroup';
export * from './ResizablePanelGroup.types';
export * from './renderResizablePanelGroup';
export * from './useResizablePanelGroup';
export * from './useResizablePanelGroupStyles.styles';
