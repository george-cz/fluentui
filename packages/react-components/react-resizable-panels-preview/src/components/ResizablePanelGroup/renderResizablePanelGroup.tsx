/** @jsxRuntime automatic */
/** @jsxImportSource @fluentui/react-jsx-runtime */

import { assertSlots } from '@fluentui/react-utilities';
import type { ResizablePanelGroupState, ResizablePanelGroupSlots } from './ResizablePanelGroup.types';

/**
 * Render the final JSX of ResizablePanelGroup
 */
export const renderResizablePanelGroup_unstable = (state: ResizablePanelGroupState) => {
  assertSlots<ResizablePanelGroupSlots>(state);

  // TODO Add additional slots in the appropriate place
  return <state.root />;
};
