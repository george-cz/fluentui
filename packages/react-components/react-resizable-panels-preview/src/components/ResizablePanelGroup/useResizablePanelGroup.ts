import * as React from 'react';
import { getIntrinsicElementProps, slot } from '@fluentui/react-utilities';
import type { ResizablePanelGroupProps, ResizablePanelGroupState } from './ResizablePanelGroup.types';

/**
 * Create the state required to render ResizablePanelGroup.
 *
 * The returned state can be modified with hooks such as useResizablePanelGroupStyles_unstable,
 * before being passed to renderResizablePanelGroup_unstable.
 *
 * @param props - props from this instance of ResizablePanelGroup
 * @param ref - reference to root HTMLDivElement of ResizablePanelGroup
 */
export const useResizablePanelGroup_unstable = (
  props: ResizablePanelGroupProps,
  ref: React.Ref<HTMLDivElement>,
): ResizablePanelGroupState => {
  return {
    // TODO add appropriate props/defaults
    components: {
      // TODO add each slot's element type or component
      root: 'div',
    },
    // TODO add appropriate slots, for example:
    // mySlot: resolveShorthand(props.mySlot),
    root: slot.always(
      getIntrinsicElementProps('div', {
        ref,
        ...props,
      }),
      { elementType: 'div' },
    ),
  };
};
