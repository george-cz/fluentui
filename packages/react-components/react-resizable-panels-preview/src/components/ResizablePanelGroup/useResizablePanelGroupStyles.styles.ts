import { makeStyles, mergeClasses } from '@griffel/react';
import type { SlotClassNames } from '@fluentui/react-utilities';
import type { ResizablePanelGroupSlots, ResizablePanelGroupState } from './ResizablePanelGroup.types';

export const resizablePanelGroupClassNames: SlotClassNames<ResizablePanelGroupSlots> = {
  root: 'fui-ResizablePanelGroup',
  // TODO: add class names for all slots on ResizablePanelGroupSlots.
  // Should be of the form `<slotName>: 'fui-ResizablePanelGroup__<slotName>`
};

/**
 * Styles for the root slot
 */
const useStyles = makeStyles({
  root: {
    // TODO Add default styles for the root element
  },

  // TODO add additional classes for different states and/or slots
});

/**
 * Apply styling to the ResizablePanelGroup slots based on the state
 */
export const useResizablePanelGroupStyles_unstable = (state: ResizablePanelGroupState): ResizablePanelGroupState => {
  const styles = useStyles();
  state.root.className = mergeClasses(resizablePanelGroupClassNames.root, styles.root, state.root.className);

  // TODO Add class names to slots, for example:
  // state.mySlot.className = mergeClasses(styles.mySlot, state.mySlot.className);

  return state;
};
