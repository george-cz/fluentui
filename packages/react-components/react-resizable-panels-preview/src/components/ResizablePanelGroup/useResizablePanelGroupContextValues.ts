import * as React from 'react';
import { ResizablePanelGroupState, ResizablePanelGroupContextValues } from './ResizablePanelGroup.types';

export function useResizablePanelGroupContextValues(state: ResizablePanelGroupState): ResizablePanelGroupContextValues {
  const { resizeState } = state;

  const panelGroupContext = React.useMemo(
    () => ({
      resizeState,
    }),
    [resizeState],
  );

  return { resizablePanelGroup: panelGroupContext };
}
