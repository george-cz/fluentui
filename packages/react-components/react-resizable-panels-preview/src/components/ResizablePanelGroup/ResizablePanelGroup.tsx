import * as React from 'react';
import type { ForwardRefComponent } from '@fluentui/react-utilities';
import { useCustomStyleHook_unstable } from '@fluentui/react-shared-contexts';
import { useResizablePanelGroup_unstable } from './useResizablePanelGroup';
import { renderResizablePanelGroup_unstable } from './renderResizablePanelGroup';
import { useResizablePanelGroupStyles_unstable } from './useResizablePanelGroupStyles.styles';
import type { ResizablePanelGroupProps } from './ResizablePanelGroup.types';

/**
 * ResizablePanelGroup component - TODO: add more docs
 */
export const ResizablePanelGroup: ForwardRefComponent<ResizablePanelGroupProps> = React.forwardRef((props, ref) => {
  const state = useResizablePanelGroup_unstable(props, ref);

  useResizablePanelGroupStyles_unstable(state);
  // TODO update types in packages/react-components/react-shared-contexts/src/CustomStyleHooksContext/CustomStyleHooksContext.ts
  // https://github.com/microsoft/fluentui/blob/master/rfcs/react-components/convergence/custom-styling.md
  useCustomStyleHook_unstable('useResizablePanelGroupStyles_unstable')(state);
  return renderResizablePanelGroup_unstable(state);
});

ResizablePanelGroup.displayName = 'ResizablePanelGroup';
