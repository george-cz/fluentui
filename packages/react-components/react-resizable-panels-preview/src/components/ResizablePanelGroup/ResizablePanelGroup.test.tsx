import * as React from 'react';
import { render } from '@testing-library/react';
import { isConformant } from '../../testing/isConformant';
import { ResizablePanelGroup } from './ResizablePanelGroup';

describe('ResizablePanelGroup', () => {
  isConformant({
    Component: ResizablePanelGroup,
    displayName: 'ResizablePanelGroup',
  });

  // TODO add more tests here, and create visual regression tests in /apps/vr-tests

  it('renders a default state', () => {
    const result = render(<ResizablePanelGroup>Default ResizablePanelGroup</ResizablePanelGroup>);
    expect(result.container).toMatchSnapshot();
  });
});
