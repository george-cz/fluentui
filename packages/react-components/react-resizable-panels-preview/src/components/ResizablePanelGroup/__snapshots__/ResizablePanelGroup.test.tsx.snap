// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResizablePanelGroup renders a default state 1`] = `
<div>
  <div
    class="fui-ResizablePanelGroup"
  >
    Default ResizablePanelGroup
  </div>
</div>
`;
